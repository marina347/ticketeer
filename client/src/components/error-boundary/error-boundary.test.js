import React from "react";
import { shallow } from "enzyme";
import ErrorBoundary from "./error-boundary.component";

describe("ErrorBoundary component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ErrorBoundary />);
  });

  it("Should render ErrorBoundary component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should  render error container if there is error", () => {
    wrapper.setState({ hasError: true });
    expect(wrapper.find('[id="error"]').exists()).toBe(true);
  });
  it("Should not render error container if there is no error", () => {
    expect(wrapper.find('[id="error"]').exists()).toBe(false);
  });
  
});
