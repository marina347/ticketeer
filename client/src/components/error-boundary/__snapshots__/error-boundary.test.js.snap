// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component Should render ErrorBoundary component 1`] = `""`;
